### PR TITLE
vexpress/sunxi refactoring

### PR DESCRIPTION
**Pull request by jforissier**
Re-factoring vexpress/sunxi platforms to kill duplicate code and prepare for HiKey support.

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
Having a new pm repository is a bit overkill. Can we have the PM files in kernel or tee repository?

**Comment by jenswi-linaro** [comment]
I agree with Pascal.
Nice to see stuff move out from `plat-*` though.

**Comment by jforissier** [comment]
Comments addressed. I have dropped the original commits since there the code is small enough to not require incremental review I think. Also note that the last commit also touches plat-stm (I have updated the commit line).

File: core/arch/arm/include/kernel/pm_stubs.h [hunk 1]
```diff
@@ -25,13 +25,13 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef PM_DEBUG_H
-#define PM_DEBUG_H
+#ifndef PM_H
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/include/kernel/pm_stubs.h]: PM_STUBS_H instead if PM_H

File: core/arch/arm/include/kernel/pm_stubs.h [hunk 2]
```diff
@@ -25,13 +25,13 @@
...
+#define PM_H
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/include/kernel/pm_stubs.h]: PM_STUBS_H instead of PM_H

**Comment by pascal-brand-st-dev** [comment]
2nd commit message: Can't it be

    "Move PM handler stubs to core/arch/arm/kernel/p……m_stubs.c"

instead of

    "stm, sunxi, vexpress: move PM handler stubs to core/arch/arm/kernel/p……m_stubs.c"?

as it is related to all the supported platforms?

**Comment by jforissier** [comment]
@pascal-brand-st-dev true, will fix. In my mind we already have hikey ;)

**Comment by pascal-brand-st-dev** [comment]
Once PM_H is fixed:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jforissier** [comment]
Review comments have been addressed.

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by jforissier** [comment]
Review commits squashed, tags applied, commit comment updated.

**Pull request merged**